pyhf/interpolate.py

```diff
@@ -53,24 +87,53 @@ def summand(down, nom, up, alpha):
 
 def _hfinterp_code1(histogramssets, alphasets):
     tensorlib, _ = get_backend()
+    i = _hfinterpolator_code1(histogramssets)
+    return i.interpolate(alphasets)
+
+class _hfinterpolator_code1(Interpolator):
+    def _precompute(self, shape=(1,1)):
+        tensorlib, _ = get_backend()
+
+        # did things change that we need to recompute?
+        tensor_type_change = tensorlib != self.tensorlib
+        shape_change = shape != self.shape
+
+        if tensor_type_change:
+            # nb: self.histogramssets will never change
+            histogramssets = tensorlib.astensor(self.histogramssets)
+            self.allset_allhisto_deltas_up = tensorlib.divide(histogramssets[:,:,2], histogramssets[:,:,1])
```

Comment:
actually we do not need `histogramssets` to be tensors of the backend, the division can be done in numpy .. the only thing that needs to be an actual tensor of the backend are  `self.X`. This will simplify the computational graph of e.g. ML backends since they will not have to know about `histogramssets` but only about the tensors that are actually used

Reply:
discuss in #290 close here